test-data/unit/check-flags.test

```diff
@@ -376,3 +376,133 @@ if y:  # E: Condition must be a boolean
 if z:  # E: Condition must be a boolean
   pass
 [builtins fixtures/bool.pyi]
+
+[case testDisallowImplicitTypesIgnoreMissingTypes]
+# flags: --ignore-missing-imports --disallow-implicit-any-types
+from missing import MyType
+
+def f(x: MyType) -> None:  # E: Argument 1 to "f" is implicitly converted to "Any" due to import from unanalyzed module
+    pass
+
+[case testDisallowImplicitTypes]
+# flags: --disallow-implicit-any-types
+from missing import MyType
+
+def f(x: MyType) -> None:
+    pass
+[out]
+main:2: error: Cannot find module named 'missing'
+main:2: note: (Perhaps setting MYPYPATH or using the "--ignore-missing-imports" flag would help)
+main:4: error: Argument 1 to "f" is implicitly converted to "Any" due to import from unanalyzed module
+
+[case testDisallowImplicitAnyVariableDefinition]
+# flags: --ignore-missing-imports --disallow-implicit-any-types
+from missing import Unchecked
+
+t: Unchecked = 12  # E: Type of variable is implicitly converted to "Any" due to import from unanalyzed module
+
+[case testDisallowImplicitAnyGeneric]
+# flags: --ignore-missing-imports --disallow-implicit-any-types
+from missing import Unchecked
+from typing import List
+
+def foo(l: List[Unchecked]) -> List[Unchecked]:
+    t = []  # type: List[Unchecked]
+    return l
+[builtins fixtures/list.pyi]
+[out]
+main:5: error: Return type is implicitly converted to List[Any] due to import from unanalyzed module
+main:5: error: Argument 1 to "foo" is implicitly converted to List[Any] due to import from unanalyzed module
+main:6: error: Type of variable is implicitly converted to List[Any] due to import from unanalyzed module
+
+[case testDisallowImplicitAnyInherit]
+# flags: --ignore-missing-imports --disallow-implicit-any-types
+from missing import Unchecked
+from typing import List
+
+class C(Unchecked): # E: Base type Unchecked is implicitly converted to "Any" due to import from unanalyzed module
+    pass
+
+class A(List[Unchecked]): # E: Base type is implicitly converted to List[Any] due to import from unanalyzed module
+    pass
+[builtins fixtures/list.pyi]
+
+[case testDisallowImplicitAnyAlias]
+# flags: --ignore-missing-imports --disallow-implicit-any-types
+from missing import Unchecked
+from typing import List
+
+X = List[Unchecked]
+
+def f(x: X) -> None:  # E: Argument 1 to "f" is implicitly converted to List[Any] due to import from unanalyzed module
+    pass
+[builtins fixtures/list.pyi]
+[case testDisallowImplicitAnyCast]
```

Comment:
Trivial: missing blank line before this line.  It's a lot harder to read the tests if there isn't a blank line before each new testcase.